src/clj_sqs_extended/core.clj

```diff
@@ -72,43 +73,58 @@
   (.deleteQueue sqs-client url))
 
 (defn send-message
-  [sqs-client url message]
-  (.sendMessage sqs-client url message))
+  ([sqs-client url body]
+   (send-message sqs-client url body {}))
+  ([sqs-client url body
+    {:keys [format]
+     :or   {format :transit}
+     :as   opts}]
+   (let [payload (serdes/serialize body format)]
+     (.sendMessage sqs-client url payload))))
 
 (defn send-fifo-message
-  [sqs-client url message group-id]
-  (let [request (SendMessageRequest. url message)]
-    ; WATCHOUT: The group ID is mandatory when sending fifo messages.
-    (doto request (.setMessageGroupId group-id))
-    (.sendMessage sqs-client request)))
+  ([sqs-client url body group-id]
+   (send-fifo-message sqs-client url body group-id {}))
+  ([sqs-client url body group-id
+    {:keys [format]
+     :or   {format :transit}
+     :as   opts}]
+   (let [payload (serdes/serialize body format)
+         request (SendMessageRequest. url payload)]
+     ; WATCHOUT: The group ID is mandatory when sending fifo messages.
+     (doto request (.setMessageGroupId group-id))
+     (.sendMessage sqs-client request))))
 
 (defn delete-message
-  [sqs-client url message]
-  (let [request (DeleteMessageRequest. url (:receiptHandle message))]
+  [sqs-client url handle]
+  (let [request (DeleteMessageRequest. url (:receiptHandle handle))]
     (.deleteMessage sqs-client request)))
 
 (defn receive-message
   ([sqs-client url]
    (receive-message sqs-client url {}))
   ([sqs-client url
-    {:keys [wait-time
+    {:keys [format
+            wait-time
             visibility-timeout
```

Comment:
can we remove the visibility-timeout setting here [as noted last time](https://github.com/Motiva-AI/clj-sqs-extended/pull/9#discussion_r461358206)? this is a configuration feature that I want to remove from this library. Or, are you leaving this here deliberately?

Reply:
Having this available on receive-message only might not interfere too much with devops and infrastructure configuration since it doesn't change the entire queue? Even though I have basically no real-life experience with sending SQS messages between real-world applications I thought this might come in handy.

Reply:
that's true. this would only change these particular messages. but from my experience, I find having this option available at two different places confusing as a user.

Reply:
"AAAND It's Gone" :wink: